snare

```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/env python3.7
+
+# Copyright (c) 2019 King's College London, created by the Software Development
+# Team <http://soft-dev.org/>
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0>, or the MIT license <LICENSE-MIT
+# or http://opensource.org/licenses/MIT>, or the UPL-1.0 license
+# <http://opensource.org/licenses/UPL> at your option. This file may not be
+# copied, modified, or distributed except according to those terms.
+
+import getopt, hashlib, hmac, json, os, subprocess, sys, tempfile, traceback
+from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler
+from urllib.parse import unquote
+
+class HookHandler(BaseHTTPRequestHandler):
+    def do_POST(self):
+        data = self.rfile.read(int(self.headers["Content-Length"]))
+        if not self._check_secret(data):
+            self.send_response(401)
+            return
+
+        data_str = unquote(data.decode("utf-8")[8:])
```

Comment:
utf-8 is default encoding, so you could kill the explicit char encoding.

https://docs.python.org/3/library/stdtypes.html?highlight=decode#bytes.decode
```
bytes.decode(encoding="utf-8", errors="strict")
```

Does the JSON github send you specify an encoding actually?

Reply:
what's the `[8:]` for?

Reply:
The string starts with `payload=` which needs stripping.